contribs/gmf/src/directives/layertree.js

```diff
@@ -758,15 +758,22 @@ gmf.LayertreeController.prototype.removeNode = function(node) {
 
 /**
  * Set the resolution of the map with the max or min resolution of the node.
- * @param {GmfThemesNode} node Layer tree node.
+ * @param {ngeo.LayertreeController} treeCtrl ngeo layertree controller, from
+ *     the current node.
  * @export
  */
-gmf.LayertreeController.prototype.zoomToResolution = function(node) {
+gmf.LayertreeController.prototype.zoomToResolution = function(treeCtrl) {
+  var node = /** @type {GmfThemesNode} */ (treeCtrl.node);
   var view = this.map.getView();
   var resolution = node.minResolutionHint || node.maxResolutionHint;
   if (resolution !== undefined) {
     view.setResolution(resolution);
   }
+
+  // automatically activate layer if not already active
+  if (this.getNodeState(treeCtrl) != 'on') {
```

Comment:
It will never be true since the only way to call zoomToResolution for now is by clicking on the zoom icon, which is only displayed if gmfLayertreeCtrl.getNodeState(layertreeCtrl) == 'on'